comp/process/runner/component.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package runner implements a component to run data collection checks in the Process Agent.
+package runner
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// team: processes
+
+// Component is the component type.
+type Component interface {
+	GetChecks() []Check
+	Run(exit <-chan struct{}) error
```

Comment:
What do you think about using [context.Context](https://pkg.go.dev/context) which is more idiomatic than  `<-chan struct{}`?

Reply:
Point taken, however here run is a daemon-like execution of the runner, so a channel would be appropriate?

Reply:
I think it is fine to use [context.Context](https://pkg.go.dev/context) for deamon-like execution (fx does).